docs/unread_messages.rst

```diff
@@ -0,0 +1,31 @@
+# Unread message synchronization
+
+In general displaying unread counts for all streams and topics may require
+downloading an unbounded number of messages. Consider a user who has a muted
+stream or topic and has not read the backlog in a month; to have an accurate
+unread count we would need to load all messages this user has received in the
+past month. This is inefficient for web clients and even more for mobile
+devices.
+
+We work around this by including a list of unread message ids in the initial
+state grouped by recipient and subject. This data is included in the
+`unread_msgs` key if both update_message_flags and message are required in the
+register call.
+
+.. code-block:: json
+    [
+        [
+            {
+                'display_recipient': 'general',
```

Comment:
The code block doesn't seem to be rendering, try using double-quotes (`"`) so the renderer can recognize the syntax.